File: src/main/kotlin/magicbook/gtlitecore/common/block/itemblocks/DimensionDisplayItemBlock.java
package magicbook.gtlitecore.common.block.itemblocks;

import codechicken.lib.model.ModelRegistryHelper;
import codechicken.lib.util.TransformUtils;
import magicbook.gtlitecore.client.renderer.IItemRendererManager;
import magicbook.gtlitecore.client.renderer.handler.DimensionDisplayItemRenderer;
import magicbook.gtlitecore.common.block.blocks.BlockDimensionDisplay;
import net.minecraft.block.Block;
import net.minecraft.client.renderer.block.model.ModelResourceLocation;
import net.minecraft.item.ItemBlock;
import net.minecraft.item.ItemStack;
import net.minecraft.util.ResourceLocation;

public class DimensionDisplayItemBlock extends ItemBlock implements IItemRendererManager
{

    public DimensionDisplayItemBlock(Block block)
    {
        super(block);
    }

    public static String getDimension(ItemStack stack)
    {
        if (stack.getItem() instanceof DimensionDisplayItemBlock)
        {
            return ((BlockDimensionDisplay) Block.getBlockFromItem(stack.getItem())).getDimension();
        }
        return null;
    }

    @Override
    public void onRendererRegistry(ResourceLocation location)
    {
        ModelRegistryHelper.register(new ModelResourceLocation(location, "inventory"),
                new DimensionDisplayItemRenderer(TransformUtils.DEFAULT_ITEM, modelRegistry ->
                        modelRegistry.getObject(new ModelResourceLocation(location, "inventory"))));
    }

}

File: src/main/kotlin/gregtechlite/gtlitecore/common/block/item/DimensionDisplayItemBlock.kt
package gregtechlite.gtlitecore.common.block.item

import codechicken.lib.model.ModelRegistryHelper
import gregtechlite.gtlitecore.client.renderer.IItemRendererManager
import gregtechlite.gtlitecore.client.renderer.handler.DimensionDisplayItemRenderer
import gregtechlite.gtlitecore.common.block.base.BlockDimensionDisplay
import net.minecraft.block.Block
import net.minecraft.client.renderer.block.model.ModelResourceLocation
import net.minecraft.item.ItemBlock
import net.minecraft.item.ItemStack
import net.minecraft.util.ResourceLocation

class DimensionDisplayItemBlock(block: Block) : ItemBlock(block), IItemRendererManager
{

    override fun onRendererRegistry(location: ResourceLocation)
    {
        val modelLocation = ModelResourceLocation(location, "inventory")
        ModelRegistryHelper.register(modelLocation, DimensionDisplayItemRenderer(modelLocation))
    }

    companion object
    {
        fun getDimension(stack: ItemStack): String?
        {
            if (stack.item is DimensionDisplayItemBlock)
            {
                return (Block.getBlockFromItem(stack.item) as BlockDimensionDisplay).dimension
            }
            return null
        }
    }

}
